### DISCUSSION TITLE
Detect size during upload

### DISCUSSION THREAD

**Discussion by SDGPeteBatin** [pionl/laravel-chunk-upload#169] [Q&A]
Is there a way I can get retrieve the size of the upload in progress while it uploads.  I'm not looking for the total finished size, just the size of the upload so far.  So if it's 50% complete, retrieve what size that is?

I've found these methods,
getBytesStart() - returns the starting bytes for current request
getBytesEnd() - returns the ending bytes for current request
getBytesTotal() - returns the total bytes for the file

But I'm not entirely sure what the first 2 would return, I'm guessing the 3 is for when the upload has completed?  Is getBytesEnd() what I'm looking for?  Current request is a little ambiguous, is that the current chunk or the current file (if multiple files).

Reason for this, I want to set a maximum upload size but I don't want to waste processes uploading a file that will ultimately fail validation if it's too large.  So if the max size is 100MB and at 50% it's already 120MB there's no point allowing the continuation of the upload and I'd like to terminate it.

**Comment by pionl** [comment]
`getBytesTotal` should return the size of the whole file, not the chunk.


I think you can throw an exception if the bytes total exceeded your limit. At this moment the package has no built in solution but PR should not be hard to do :)

**Reply by SDGPeteBatin** [reply]
So is there currently a way to get the size of the current progress?

My code pretty much looks like this at the moment https://github.com/pionl/laravel-chunk-upload-example/blob/master/example/src/Http/Controllers/UploadController.php

What's stopping someone from uploading a huge file or multiple huge files from different IPs and filling up the disk?  

At the moment the only opportunity for size detection I can see is when files have completed uploading.  I'm using $file->getSize(); in the saveFile method but to trigger that I would have already had to allow a file larger than I want to be uploaded to the server.

I need somehow to be able to get the total size of collected chunks at every execution of either $save = $receiver->receive() or $save->isFinished() and then to be able to end the process (and remove chunks)......I'm just not Jedi enough for it yet.
